### ISSUE TITLE
Grafana Login Timeout (v9.2.0)

### ISSUE THREAD

**Issue by fatalbyte** [grafana/grafana#56994]
**What happened**:
Upgraded to v9.2.0

**What you expected to happen**:
No change in login / timeout behavior.

**How to reproduce it (as minimally and precisely as possible)**:
After upgrading to v9.2.0 Grafana will log out of the active user account in approximately 60 seconds.

**Anything else we need to know?**:
I did review the Grafana setting and preferences and did not see any option to increase the timeout length, or to set it to infinite. 

**Environment**:
- Grafana version: 9.2.0
- Data source type & version: InfluxDB
- OS Grafana is installed on: Raspberry Pi 10 (32-bit)
- User OS & Browser: Mac OSX / Chrome
- Grafana plugins: Telegraf
- Others: NA

**Comment by grafanabot** [comment]
Thanks for creating this issue! We think it's missing some basic information. 

Follow the issue template and add additional information that will help us replicate the problem. 
For data visualization issues: 
- Query results from the inspect drawer (data tab & query inspector)
- Panel settings can be extracted in the panel inspect drawer JSON tab

For dashboard related issues: 
- Dashboard JSON can be found in the dashboard settings JSON model view

For authentication, provisioning and alerting issues, Grafana server logs are useful. 

Happy graphing!

**Comment by zatie12** [comment]
Since upgrading to 9.2.0 (Ubuntu) I am regularly getting kicked to the login screen, usually after about 5 to 10 minutes. I've gone back to 9.1.8 for the time being.

**Comment by Pimmetje** [comment]
I have complaints from users that they get logged out at random. I downgraded to 9.1.8

**Comment by hudeldudel** [comment]
I see a lot of user token not found after cleanup jobs.
Example:
logger=cleanup t=2022-10-13T08:49:17.257133824Z level=info msg="Completed cleanup jobs" duration=30.036723ms
logger=context t=2022-10-13T08:50:01.984095009Z level=warn msg="Failed to look up user based on cookie" error="user token not found"

**Comment by mckn** [comment]
Duplicate of https://github.com/grafana/grafana/issues/57485

**Comment by fatalbyte** [comment]
> I had this issue happening after upgrades and the fix appears to be cleaning the contents of `user_auth_token` table. Mine contained a few entries for the same user and I suspect that the query was hitting the first result which was already stale and so executing a login timeout that many people are experiencing here.
> 
> Just delete the contents of that table in the grafana.db and login again.

Good to know!  What's the easiest way to do that?

**Comment by gdbinit** [comment]
In case of Ubuntu something like:
```bash
sudo sqlite3 /var/lib/grafana/grafana.db "delete * from user_auth_token;" ""
```

For other distros just find where `grafana.db` file is and adjust the path in above's command.

**Comment by blastehh** [comment]
> 

Thanks, trying this out.
The command for deleting all rows in sqlite is actually
```
sqlite3 /var/lib/grafana/grafana.db "delete from user_auth_token;"
```

**Comment by pdp455** [comment]
I am facing similar issue as well after upgrading to 9.2.3
'user token not found' errors started after an upgrade to 9.2.3

**Comment by hudeldudel** [comment]
Tested again with 9.2.4 and deleting the content from user_auth_token table.
Didn't help permanently for me. Logouts appear again after some time.

**Comment by alexandzors** [comment]
I just lost an unsaved Dashboard (that i was making edits to) because I walked away for a few minutes and Grafana forced logged the admin user out. On v9.2.6.

**Comment by metanull-operator** [comment]
Using 9.2.3 and getting logged out much more quickly than in the past. The amount of time between last login and being logged out is inconsistent. Sometimes an hour, sometimes a day. It used to be weeks. I have also lost unsaved dashboards due to this issue.

**Comment by metanull-operator** [comment]
For me, all three of those values are commented out in my configuration file. They should be using the defaults.

As another incident report. I was logged out while editing a dashboard this morning. I logged back in and was logged out by the system not more than 20 minutes later, again losing some work.

**Comment by Pimmetje** [comment]
Due to https://grafana.com/blog/2022/11/29/grafana-security-release-new-versions-with-high-severity-security-fix-for-cve-2022-31097/ it's no longer a option to stay on 9.1.8. But this issue still seem to exist in the latest version.

**Comment by mgyongyosi** [comment]
By any chance could you please provide more logs (around the time when the logout happens), possibly a HAR file? Does this happen with forms authentication (username/password) or with other identity providers?

To enable debug logging please add the following to Grafana's configuration file (`grafana.ini`/`custom.ini`):
```ini
[log]
level = debug
```
Thanks!

**Comment by zatie12** [comment]
I'm still getting logged out in 9.3.1. Reverted to 9.1.8.

**Comment by Pimmetje** [comment]
I have normal password authentication

**Comment by metanull-operator** [comment]
On one system of mine this has escalated to being logged out every 20-40 minutes.

My logs have a lot of messages about auth token rotations but those may be normal. The lines that appear to be most relevant are:

```
Dec 15 13:44:13 nuc04 grafana-server[558296]: logger=context t=2022-12-15T13:44:13.895542863-07:00 level=warn msg="Failed to look up user based on cookie" error="user token not found"
Dec 15 13:44:13 nuc04 grafana-server[558296]: logger=context userId=0 orgId=0 uname= t=2022-12-15T13:44:13.895779453-07:00 level=debug msg="Expiring invalid cookie"
```

If you need more logs, see attached. I have removed "Sending Query" lines from the logs. 

I can attempt to provide an HAR file another time.

[grafana-logs-2022-12-15.txt](https://github.com/grafana/grafana/files/10240608/grafana-logs-2022-12-15.txt)

**Comment by metanull-operator** [comment]
Adding: Using forms authentication.

I have never generated a HAR file and will have to look into that. Suggestions on good tutorials?

**Comment by Pimmetje** [comment]
U can generate a har from de debug console of your browser. (right click on a request in the network-tab).

**Comment by hasso** [comment]
This seems to happen only with users who use multiple machines/browsers (therefore sessions) in parallel.

**Comment by cburghardt** [comment]
I do have the same issue since upgrading to version 9. Only 1 session is 
active and I am being logged out several times per day.

Am 16. Januar 2023 14:39:13 schrieb Hasso Tepper ***@***.***>:

**Comment by chriscareycode** [comment]
Having this issue also as well. After upgrade > 9.2.0 my dashboards will log out randomly throughout the day. A very basic OSS Grafana install with default auth. I rolled back to 9.1.7 (the last known good I had handy), and they all stay logged in again.

**Comment by elektronikworkshop** [comment]
Getting logged out too. Used to be weeks now its not even a day.

**Comment by metanull-operator** [comment]
I do not necessarily think multiple sessions is a factor here. I generally use the same system and browser for all access to my dashboard. With this bug I can be logged out of my Grafana sessions 4-6 times during a long working session, and not once will I have accessed the same Grafana instance from any other browser or system in the meantime. Sometimes when I am inexplicably logged out I know I haven’t accessed the same Grafana instance from any other browser in weeks. 

I am the only user of my Grafana instance.

**Comment by mckn** [comment]
@mgyongyosi just want to check that you are working on this? Feels like an issue a lot of people are experiencing.

**Comment by mgyongyosi** [comment]
@mckn We can't reproduce the issue, but we have a hunch what's causing it. We are working on a patch for 9.4.

**Comment by jorgegene** [comment]
We are facing the same problem on Grafana 9.3.0.
OAuth configured and users can't use the tool cause they get kicked after 5 min from login.
Is there any temporal solution so the users can work without being logged out?

**Comment by chriscareycode** [comment]
Reporting back here that I upgraded from 9.1.7 to 9.4.1 and it appears to have fixed the logout issue for me so far. Thank you!

**Comment by IgorOhrimenko** [comment]
Some time ago I fixed the problem by `sudo sqlite3 /var/lib/grafana/grafana.db "delete from user_auth_token;"`
Yesterday I upgraded 9.3.6 -> 9.4.3 and issue is reverted.
Today I again ran `sudo sqlite3 /var/lib/grafana/grafana.db "delete from user_auth_token;"`. I hope this trick is still working.

Reopen?

**Comment by IgorOhrimenko** [comment]
The trick does not work anymore. :@

**Comment by Normand-Nadon** [comment]
I am adding my voice to this issue, It is in fact still a problem.
We have set the login timeouts to 12 months as a test for users authenticating internally to Grafana (some admin users use OAuth, but "viewer" users connect internally)

The sessions are failing at random intervals, never lasting a full day...
There is an error message on screen
![image](https://user-images.githubusercontent.com/60982016/229567331-5d1303ac-c447-453b-abd7-369cb280aa46.png)

**Comment by Normand-Nadon** [comment]
@mgyongyosi can this issue be reopened ?

Should we create a new issue for the subject?

**Comment by hasso** [comment]
+1 Still hapens here.

**Comment by zatie12** [comment]
I am still getting logged out, maybe once or twice per day

**Comment by Pimmetje** [comment]
Here also still issues, it's a bit less but it still exists.

**Comment by mckn** [comment]
> @mckn We can't reproduce the issue, but we have a hunch what's causing it. We are working on a patch for 9.4.

Got it but since there seems to be a lot of users experiencing this issue it feels like something we should prioritize fixing.

**Comment by mgyongyosi** [comment]
@mckn I'm gonna open a PR soon that would fix this soon and we'll backport it to 9.2, 9.3 an 9.4.
